server/lorax_server/models/flash_causal_lm.py

```diff
@@ -1326,24 +1346,17 @@ def warmup(self, batch: FlashCausalLMBatch, max_new_tokens: int, embedding_model
             # Estimate the memory overhead from CUDA graphs so we can subtract it from the kv cache.
             # Needs to be estimated here rather than fully initialized as the graph cache relies on the
             # cache manager being set.
-            self.model_graph_wrapper = GraphCache(
-                self.model,
-                self.device,
-                self.kv_cache,
-                self.adapter_layers,
-                self.traced_adapter_layers,
-                self._forward_context,
-                max_total_tokens,
-                self.num_heads,
-                self.num_kv_heads,
-                self.sliding_window_blocks,
-                self.layer_to_lora_weights,
-                self.punica_wrapper,
-            )
+            self.init_graph_wrapper(max_total_tokens)
             graph_cache_memory = self.model_graph_wrapper.get_estimated_cache_memory()
             logger.info("Estimated graph cache memory: {} MB", graph_cache_memory / 1024 / 1024)
             torch.cuda.synchronize(self.device)
 
+        logger.info(f'Post warmup cuda memory: {get_cuda_free_memory(self.device, 1) / (1024 ** 3)}')
+        del batch
```

Comment:
Hmm, I don't think you can delete the batch at this point, as it's needed to estimate the prefill memory overhead. I would compare with how TGI does it.

Reply:
Okay they seem to delete the batch [after `num_block` calculation](https://github.com/huggingface/text-generation-inference/blob/856709d5c3d24a9cad53c2ab94835c6190bacf3c/server/text_generation_server/models/flash_causal_lm.py#L1609) as well, so I'll do the same.

Reply:
Basically, I'll revert back to whatever we had.